### ISSUE TITLE
Value Error with small n_atoms_max in matrix descriptor

### ISSUE THREAD

**Issue by resnant** [SINGROUP/dscribe#53]
Hello,
I encountered an issue related to `matrixdescriptor` class.

# Summary
In short, `ValueError: index can't contain negative values` occurs when `n_atoms_max` is smaller than the input structure.

This error comes from this line ([matrixdescriptor.py#L192](https://github.com/SINGROUP/dscribe/blob/master/dscribe/descriptors/matrixdescriptor.py#L192)):
`padded = np.pad(array, [(0, self.n_atoms_max-n_atoms)]*n_dim, 'constant')`

I guess we have two options to address this issue:
1. Adding an assertion or warning to ensure `n_atoms <= n_atoms_max `
2. Automatically choosing `n_atoms_max` depending on the input structure:  \
matminer does this in [their implementation of Sine Coulomb Matrix](https://github.com/hackingmaterials/matminer/blob/3b3d117d106931a8a8053b1dd716eccba6616181/matminer/featurizers/structure.py#L698)

Do you have any ideas about this issue?

# To reproduce
```
from ase.build import bulk
from dscribe.descriptors import EwaldSumMatrix, SineMatrix

esm = EwaldSumMatrix(n_atoms_max=1, flatten=False)
# NaCl crystal created as an ASE.Atoms
nacl = bulk("NaCl", "rocksalt", a=5.64)

esm.create(nacl, n_jobs=1)

# We get the same error with Sine Matrix as well
# scm = SineMatrix(n_atoms_max=1)
# scm.create(nacl)
```

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-3b115864245f> in <module>
----> 1 esm.create(nacl, n_jobs=1)

/opt/conda/lib/python3.7/site-packages/dscribe/descriptors/ewaldsummatrix.py in create(self, system, accuracy, w, rcut, gcut, a, n_jobs, verbose)
    106         # If single system given, skip the parallelization
    107         if isinstance(system, (Atoms, System)):
--> 108             return self.create_single(system, accuracy, w, rcut, gcut, a)
    109         else:
    110             self._check_system_list(system)

/opt/conda/lib/python3.7/site-packages/dscribe/descriptors/ewaldsummatrix.py in create_single(self, system, accuracy, w, rcut, gcut, a)
    187         self.rcut = rcut
    188 
--> 189         matrix = super().create_single(system)
    190         return matrix
    191 

/opt/conda/lib/python3.7/site-packages/dscribe/descriptors/matrixdescriptor.py in create_single(self, system)
    126 
    127         # Add zero padding
--> 128         matrix = self.zero_pad(matrix)
    129 
    130         # Flatten

/opt/conda/lib/python3.7/site-packages/dscribe/descriptors/matrixdescriptor.py in zero_pad(self, array)
    190         n_atoms = array.shape[0]
    191         n_dim = array.ndim
--> 192         padded = np.pad(array, [(0, self.n_atoms_max-n_atoms)]*n_dim, 'constant')
    193 
    194         return padded

<__array_function__ internals> in pad(*args, **kwargs)

/opt/conda/lib/python3.7/site-packages/numpy/lib/arraypad.py in pad(array, pad_width, mode, **kwargs)
    746 
    747     # Broadcast to shape (array.ndim, 2)
--> 748     pad_width = _as_pairs(pad_width, array.ndim, as_index=True)
    749 
    750     if callable(mode):

/opt/conda/lib/python3.7/site-packages/numpy/lib/arraypad.py in _as_pairs(x, ndim, as_index)
    517 
    518     if as_index and x.min() < 0:
--> 519         raise ValueError("index can't contain negative values")
    520 
    521     # Converting the array with `tolist` seems to improve performance

ValueError: index can't contain negative values
```

**Comment by lauri-codes** [comment]
Hi @resnant,

This is a good point. Option 2 is out of the question, as typically people will want to prefix the matrix size, as the number of features needs to be consistent across a larger dataset. I think that option 1 will be the way to go. We should raise a more informative ValueError if `n_atoms > n_atoms_max`.

Thanks for your input, I can implement this check and it will be included in the next release.

**Comment by resnant** [comment]
Hi @lauri-codes,

I agree with your plan.
Thank you for maintaining this cool project! 😄

**Comment by lauri-codes** [comment]
Now fixed in 0.5.0rc1.
